contracts/XYZSwapPair.sol

```diff
@@ -38,46 +54,63 @@ contract XYZSwapPair is IXYZSwapPair, ERC20Permit, ReentrancyGuard, VolumeTrendR
         uint256 amount1In,
         uint256 amount0Out,
         uint256 amount1Out,
-        address indexed to
+        address indexed to,
+        uint256 feeInPrecision
     );
-    event Sync(uint112 reserve0, uint112 reserve1);
+    event Sync(uint256 vReserve0, uint256 vReserve1, uint256 reserve0, uint256 reserve1);
 
     constructor() public ERC20Permit("XYZSwap LP", "XYZ-LP", "1") VolumeTrendRecorder(0) {
         factory = msg.sender;
     }
 
     // called once by the factory at time of deployment
-    function initialize(IERC20 _token0, IERC20 _token1) external override {
+    function initialize(
+        IERC20 _token0,
+        IERC20 _token1,
+        uint32 _ampBps
+    ) external {
         require(msg.sender == factory, "XYZSwap: FORBIDDEN"); // sufficient check
         token0 = _token0;
         token1 = _token1;
+        ampBps = _ampBps;
     }
 
     /// @dev this low-level function should be called from a contract
     ///                 which performs important safety checks
     function mint(address to) external override nonReentrant returns (uint256 liquidity) {
-        (uint112 _reserve0, uint112 _reserve1, ) = getReserves(); // gas savings
-        uint256 balance0 = token0.balanceOf(address(this));
-        uint256 balance1 = token1.balanceOf(address(this));
-        uint256 amount0 = balance0.sub(_reserve0);
-        uint256 amount1 = balance1.sub(_reserve1);
+        (bool isAmpPool, ReserveData memory data) = getReservesData();
+        ReserveData memory _data;
```

Comment:
The reason uniswap use `_reserve0` is avoiding shadowing. However `reserve0` and `reserve1` is inside struct `ReserveData`, so there is no-need to use `data._reserve0`.

Reply:
Yes I know, just something to consider. When I was looking at the code, because of how the underscores are "swapped", it became a little bit difficult to make sure that there was no mix-up.

For example,
```
uint256 amount0 = balance0.sub(_reserve0);
``` 
becomes
```
uint256 amount0 = _data.reserve0.sub(data.reserve0);
```

It would be easier to ensure the behaviour is similar if it was 
```
uint256 amount0 = data.reserve0.sub(_data.reserve0);
```

Reply:
Hence my suggestion is to exchange the `data` and `_data` variables